src/components/users/signin-social/signinSocialModelBinder.ts

```diff
@@ -16,7 +16,15 @@ export class SigninSocialModelBinder implements IModelBinder<SigninSocialModel>
     public async contractToModel(contract: SigninSocialContract): Promise<SigninSocialModel> {
         const model = new SigninSocialModel();
 
-        model.roles = contract.roles || [BuiltInRoles.everyone.key];
+        if (contract.roles) { // migration.
+            contract.security = {
+                roles: contract.roles
+            }
+
+            contract.roles = null;
+        }
+
+        model.security = contract.security ?? { roles: [BuiltInRoles.everyone.key] };
```

Comment:
{ roles: [BuiltInRoles.everyone.key] } is duplicated below on line 49 as well, we can const it as initialSecurity somewhere above and use it.

Reply:
done